src/Model/DAOs/LdapUserDao.php

```diff
@@ -51,15 +51,18 @@ public function __construct(Ldap $ldap, $schema)
     public function getUserByCredentials($login, $password)
     {
         try {
-            $this->ldap->bind("uid=".AbstractFilter::escapeValue($login).",".($this->schema?$this->schema.',':'').LDAP_BASEDN, $password);
+            $this->ldap->bind();
+            $user = $this->getUserByLogin($login);
+            if ($user !== null) {
+                $this->ldap->bind("uid=" . AbstractFilter::escapeValue($user->getLogin()) . "," . ($this->schema ? $this->schema . ',' : '') . LDAP_BASEDN, $password);
+            }
         } catch (LdapException $exception) {
             if ($exception->getCode() == 49) {
                 return null;
             } else {
                 throw $exception;
             }
         }
-        $user = $this->getUserByLogin($login);
         return $user;
```

Comment:
$user is possibly null, it has not been initialized, you should initialize it then either return null or throw an exception if you could not connect (most propably return null)